instrumentation/httpclient-5.0/src/main/java/com/nr/agent/instrumentation/httpclient50/WrappedFutureCallback.java

```diff
@@ -33,7 +35,14 @@ public WrappedFutureCallback (HttpRequest request, FutureCallback origCallback,
     @Override
     public void completed(T response) {
         try {
-            InstrumentationUtils.processResponse(request.getUri(), (BasicHttpResponse)response, segment);
+            if (response instanceof HttpResponse) {
+                InstrumentationUtils.processResponse(request.getUri(), (HttpResponse) response, segment);
+            } else if (response instanceof Message && ((Message)response).getHead() instanceof HttpResponse) {
+                HttpResponse resp2 = (HttpResponse)(((Message)response).getHead());
+                InstrumentationUtils.processResponse(request.getUri(), resp2, segment);
+            } else {
+                AgentBridge.getAgent().getLogger().log(Level.FINER, "Unhandled response type: "+response.getClass());
```

Comment:
This creates a StringBuilder even if the message is not going to be logged.

Could you use:
```
 AgentBridge...log(Level.FINER, "Unhandled response type: {0}", response.getClass());
```
or guard with a check to whether FINER level is enabled.

And regardless, if response is null here, it will throw a NPE.